<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>ont_artic_pipeline_DRAFT</title>
  <style>
    html {
      line-height: 1.5;
      font-family: Georgia, serif;
      font-size: 20px;
      color: #1a1a1a;
      background-color: #fdfdfd;
    }
    body {
      margin: 0 auto;
      max-width: 36em;
      padding-left: 50px;
      padding-right: 50px;
      padding-top: 50px;
      padding-bottom: 50px;
      hyphens: auto;
      word-wrap: break-word;
      text-rendering: optimizeLegibility;
      font-kerning: normal;
    }
    @media (max-width: 600px) {
      body {
        font-size: 0.9em;
        padding: 1em;
      }
    }
    @media print {
      body {
        background-color: transparent;
        color: black;
        font-size: 12pt;
      }
      p, h2, h3 {
        orphans: 3;
        widows: 3;
      }
      h2, h3, h4 {
        page-break-after: avoid;
      }
    }
    p {
      margin: 1em 0;
    }
    a {
      color: #1a1a1a;
    }
    a:visited {
      color: #1a1a1a;
    }
    img {
      max-width: 100%;
    }
    h1, h2, h3, h4, h5, h6 {
      margin-top: 1.4em;
    }
    h5, h6 {
      font-size: 1em;
      font-style: italic;
    }
    h6 {
      font-weight: normal;
    }
    ol, ul {
      padding-left: 1.7em;
      margin-top: 1em;
    }
    li > ol, li > ul {
      margin-top: 0;
    }
    blockquote {
      margin: 1em 0 1em 1.7em;
      padding-left: 1em;
      border-left: 2px solid #e6e6e6;
      color: #606060;
    }
    code {
      font-family: Menlo, Monaco, 'Lucida Console', Consolas, monospace;
      font-size: 85%;
      margin: 0;
    }
    pre {
      margin: 1em 0;
      overflow: auto;
    }
    pre code {
      padding: 0;
      overflow: visible;
    }
    .sourceCode {
     background-color: transparent;
     overflow: visible;
    }
    hr {
      background-color: #1a1a1a;
      border: none;
      height: 1px;
      margin: 1em 0;
    }
    table {
      margin: 1em 0;
      border-collapse: collapse;
      width: 100%;
      overflow-x: auto;
      display: block;
      font-variant-numeric: lining-nums tabular-nums;
    }
    table caption {
      margin-bottom: 0.75em;
    }
    tbody {
      margin-top: 0.5em;
      border-top: 1px solid #1a1a1a;
      border-bottom: 1px solid #1a1a1a;
    }
    th {
      border-top: 1px solid #1a1a1a;
      padding: 0.25em 0.5em 0.25em 0.5em;
    }
    td {
      padding: 0.125em 0.5em 0.25em 0.5em;
    }
    header {
      margin-bottom: 4em;
      text-align: center;
    }
    #TOC li {
      list-style: none;
    }
    #TOC a:not(:hover) {
      text-decoration: none;
    }
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<h1 id="artic-sars-cov-2-ont-pipeline">ARTIC SARS-CoV-2 ONT Pipeline</h1>
<h2 id="background">Background</h2>
<p>An introduction to amplicon sequencing bioinformatics can be found <a href="https://artic.network/quick-guide-to-tiling-amplicon-sequencing-bioinformatics.html">here</a>. Information regarding the core pipeline can be found <a href="https://artic.readthedocs.io/en/latest/minion/">here</a>.</p>
<p>Installation and procedure have been adapted from <a href="https://artic.network/ncov-2019/ncov2019-it-setup.html">ARTIC Network</a>.</p>
<h2 id="starting-from-basecalled-demuxed-reads">Starting from basecalled, demuxed reads</h2>
<p>This protocol is for use with <strong>basecalled</strong> AND <strong>demultiplexed</strong> reads. For further information regarding basecalling and demultiplexing of amplicon reads please refer to the ARTIC Network’s bioinformatics <a href="https://artic.network/ncov-2019/ncov2019-bioinformatics-sop.html">SOP</a>.</p>
<h3 id="input-requirements">Input requirements</h3>
<p>This protocol requries the file inputs: * A comma separated <code>sample_sheet.txt</code> in the format <code>ID,Species,Barcode,RunID</code> &gt; Note: barcodes must be in the format <code>barcode##</code>.</p>
<blockquote>
<p>Suggestions: Species is not needed for the pipeline, can be omitted. An <code>amplicon_version</code> column would be helpful. * <a href="#Batch-concatenating-FASTQ-files">Concatenated FASTQ file</a> of each barcoded sample named <code>${RunID}_${Barcode}.fastq</code></p>
</blockquote>
<h3 id="activate-the-artic-environment">Activate the ARTIC Environment</h3>
<p>Activate Conda:</p>
<pre><code>conda activate</code></pre>
<p>Activate the <code>artic-ncov2019-medaka</code> environment:</p>
<pre><code>conda activate artic-ncov2019-medaka</code></pre>
<h3 id="setup-the-project-directory">Setup the project directory</h3>
<p>Create the following project directory structure and navigate to the project folder.</p>
<pre><code>mkdir ${RunID} ${RunID}/input
cd ${RunID}</code></pre>
<h3 id="batch-execute-all-barcoded-samples">Batch execute all barcoded samples</h3>
<pre><code>sed 1d sample_sheet.txt | while IFS=, read ID Species Barcode RunID; \
    do mkdir ${ID}; \
    artic minion --medaka --normalise 200 --threads 72 \
    --scheme-directory ~/artic-ncov2019/primer_schemes \
    --read-file input/${RunID}_${Barcode}.fastq \
    nCoV-2019/V1200 \
    ${ID}/${ID}; \
done</code></pre>
<h4 id="key-output-files">Key output files</h4>
<ul>
<li><code>${ID}.rg.primertrimmed.bam</code> - BAM file for visualisation after primer-binding site trimming</li>
<li><code>${ID}.trimmed.bam</code> - BAM file with the primers left on (used in variant calling)</li>
<li><code>${ID}.merged.vcf</code> - all detected variants in VCF format</li>
<li><code>${ID}.pass.vcf</code> - detected variants in VCF format passing quality filter</li>
<li><code>${ID}.fail.vcf</code> - detected variants in VCF format failing quality filter</li>
<li><code>${ID}.primers.vcf</code> - detected variants falling in primer-binding regions</li>
<li><code>${ID}.consensus.fasta</code> - consensus sequence</li>
</ul>
<h1 id="section"></h1>
<h2 id="additional-instructions">Additional instructions</h2>
<h3 id="batch-concatenating-fastq-files">Batch concatenating FASTQ files</h3>
<p>During an ONT sequencing run, reads are written to file in batches (default = 4000), leading to multiple FASTQ files for each sample at the end of the run. The input requirement for <code>artic minion</code> is one FASTQ file per barcode named <code>${RunID}_${Barcode}.fastq</code>.</p>
<p>Follow these steps to concatenate all samples from the run at once:</p>
<pre><code># Convert sample sheet to csv (if necessary)

in2csv sample_sheet.xlxs &gt; sample_sheet.txt

# Set the PATH to the FASTQ directory

FASTQ_PATH=&quot;/PATH_TO/fastq_pass&quot;

# Concatenate FASTQ files, copy to local input/

sed 1d sample_sheet.txt | while IFS=, read ID Species Barcode RunID; \
    do cat ${FASTQ_PATH}/${Barcode}/*.fastq &gt; input/${RunID}_${Barcode}.fastq; \
done</code></pre>
<h3 id="adding-custom-primer-schemes">Adding custom primer schemes</h3>
<p>Files associated with Nikki Freed’s V1200 amplicon scheme can be downloaded <a href="https://drive.google.com/file/d/1Y85hs7-HWSsiZxU6Gh2qcij2xkW7HbZ-/view">here</a>.</p>
<p>The following files must be in the new primer scheme folder, e.g. <code>~/artic-ncov2019/primer_schemes/nCoV-2019/V1200</code>: * <code>nCoV-2019.bed</code> * <code>nCoV-2019.reference.fasta</code> * <code>nCoV-2019.reference.fasta.fai</code> * <code>nCoV-2019.scheme.bed</code> * <code>nCoV-2019.tsv</code></p>
</body>
</html>
